libraries/AP_Scripting/applets/ScriptingSwitcher/Script_Controller.md

```diff
@@ -1,15 +1,21 @@
+This script allows the user to have groups of scripts in three subdirectories of the main scripting directory and copy them into the main directory, removing any files in that directory with the .lua extension (except itself) and issue a scripting restart based on the position of a switch with the auxiliary function of "302". This provides  means to have differing scripting depending on the mission or conditions, and use whichever set is desired without having them consume memory all the time and without requiring a GCS to manage the content of the scripts directory.
+
+# Optional Mission and Parameter switching
+
+It is also possible to switch and load a different mission or set of parameters to accompany the new set of scripts at the same time. Simply include a copy of Mission_Controller.lua and/or Param_Controller.lua in each subdirectory that you desire the mission or some parameters changed along with the new scripts to be executed. 
 
 # Setup and Operation
 
-The user must setup an RCx_OPTION to function 302 to determine which of the three scripts subdirectories will be used: LOW:scripts/1, MIDDLE:scripts/2, or HIGH:scripts/3. If no RC has been established during ground start, it will behave as if subdirectory 1 is selected. Changing this switch position either prior to ground start or after, will remove ALL the *.lua files in the scripts directory (except itself), and copy any *.lua files (only) from the desginated subdirectory, and then issue a scripting restart. 
+The user must setup an RCx_OPTION to function 302 to determine which of the three scripts subdirectories will be used: LOW:scripts/1, MIDDLE:scripts/2, or HIGH:scripts/3. If no RC has been established during ground start, it will behave as if subdirectory 1 is selected. Changing this switch position either prior to ground start or after, will remove ALL the *.lua files in the scripts directory (except itself), and copy any *.lua files (only) from the designated subdirectory, and then issue a scripting restart. 
+
+If the mission or parameter script options are used for a subdirectory the mission file to be loaded at the same time that subdirectory's scripts are loaded should be named "mission.txt" and/or "params.param".
 
 # CAUTIONS
 
 If parameters are being created and used be sure that different scripts do not use the same PARAMETER KEY to create their param tables, unless you desire to use the same parameter set in two different scripts. Otherwise, param values will be corrupt when you change scripts.
```

Comment:
I think this only applies if parameters are being created am I right? It would not apply to using existing parameters (which is one way to read this). This could be confusing to some users.
Perhaps be very clear about this"If the script creates and uses **new** parameters" and maybe even add a note at the end: - "this would not apply to using existing and/or standard parameters, only to new parameters created by the script"

Reply:
I think "If parameters are being created and used" is pretty clear..the warning is about PARAM keys and is information only someone writing scripts would be concerned with or know about..."and used" is superfluous...will drop that...

Reply:
Yes that makes it clearer - it was "and used" that was confusing.